src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -316,8 +320,45 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
         }
     }
 
+    @DataProvider
+    Object[][] deepDataProvider() {
+        return new Object[][]{{10}, {50}, {100}, {500}, {1000}, {5000}, {10000}, {50000}};
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepData(int counts) {
```

Comment:
I feel like `testCanHandleDeepData` and `testCanHandleDeepDataHPFromContamination` could be refactored slightly more, considering they should be basically doing exactly the same thing.

Reply:
👍